pipeline/transformer/transformer_common/src/main/scala/weco/catalogue/transformer/WorkComparison.scala

```diff
@@ -0,0 +1,30 @@
+package weco.catalogue.transformer
+
+import uk.ac.wellcome.models.work.internal.Work
+import uk.ac.wellcome.models.work.internal.WorkState.Source
+
+object WorkComparison {
+
+  /** Returns true if `work` is:
+    *
+    *    - newer than `storedWork`
+    *    - meaningfully different to `storedWork`
+    *
+    */
+  def isNewer(storedWork: Work[Source], work: Work[Source]): Boolean = {
+    assert(
+      storedWork.id == work.id,
+      s"Cannot compare two works with different IDs: ${storedWork.id} != ${work.id}"
+    )
+
+    // If the stored work has a higher version, then it automatically wins.
+    if (storedWork.version > work.version) {
+      false
+    }
```

Comment:
:thinking: I am a little hazy on remembering where this version actually comes from, but doesn't it get bumped every time it is ingested into the adapter, and thus will be greater even when the data is the same (such as during a reharvest)?

Reply:
Yeah, so I don’t think with this branch very often in practice

Reply:
Ah I misread the original code, I get it now

Reply:
I’m going to put some more comments in this code tomorrow, because it’s clearly not obvious right now.